portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,168 +49,203 @@
+ 
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(blogMessageEntity);
     }
-    putBlogsInDatastore(postTag, message, nickname, messageReplies, parentID);
-
-    // To get the recently posted message, add its tag to the |tagsToSearch| list.
-    // If the list has tags already, clear them before adding the tag.
-    // Our goal is to get the recently posted message.
-    tagsToSearch.clear();
-    tagsToSearch.add(postTag);
-    
-    // We want to load only the recent post.
-    numberOfCommentsToDisplay = 1; 
-    
-    // Now we call |doGet| to load and respond with the message.
-    // |doGet| passes the |tagsToSearch| and |numberOfCommentsToDisplay| to the 
-    // BlogHashMap's getMessages method, which responds with the recent post.
-    doGet(request, response);
   }
-
-    // Returns a URL that points to the uploaded file, or null if the user didn't upload a file.
-    private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
-      BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
-      Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
-      List<BlobKey> blobKeys = blobs.get(formInputElementName);
-
-      if (blobKeys == null || blobKeys.isEmpty()) {
-        return null;
-      }
-
-      // Our form only contains a single file input, so get the first index.
-      BlobKey blobKey = blobKeys.get(0);
-
-      // User submitted form without selecting a file, so we can't get a URL. (live server)
-      BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
-      if (blobInfo.getSize() == 0) {
-        blobstoreService.delete(blobKey);
-        return null;
-      }
-
-      // We could check the validity of the file here, e.g. to make sure it's an image file
-      // https://stackoverflow.com/q/10779564/873165
-
-      // Use ImagesService to get a URL that points to the uploaded file.
-      ImagesService imagesService = ImagesServiceFactory.getImagesService();
-      ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);
-
-      // To support running in Google Cloud Shell with AppEngine's dev server, we must use the relative
-      // path to the image, rather than the path returned by imagesService which contains a host.
-      try {
-        URL url = new URL(imagesService.getServingUrl(options));
-        return url.getPath();
-      } catch (MalformedURLException e) {
-        return imagesService.getServingUrl(options);
+ 
+  // HELPER FUNCTION:................................................................
+  // Loads all BlogMessages from Datastore if true is passed,
+  // Otherwise if false is passed only the recent post is loaded.
+  // This is useful because each time a user posts, we only load the last BlogMessage
+  // Which reduces the time to load datastore.
+  private List<BlogMessage> LoadAllBlogsOrLast(boolean all) {
```

Comment:
To some degree, I think we should even further break apart this method into loadAllBlogs() & loadMostRecentBlogs(). This right now mix the two with an "all", which makes the underlying logic branched with a bool. If you really want to reuse the part (line 183-194), maybe add another getBlogMessageFromEntity().